src/main/java/org/gwtproject/tools/cldr/CurrencyListProcessor.java

```diff
@@ -0,0 +1,366 @@
+                    .returns(CurrencyData.class)
+                    .addCode(currencyInfo.getDefaultImplCode())
+                    .build();
+
+            currencyListBuilder
+                    .addMethod(defaultMethod);
+
+        }
+
+        if (currencies.length > 0) {
+            writeCurrencyMethod(currencies, allCurrencyData)
+                    .ifPresent(currencyListBuilder::addMethod);
+
+            writeNamesMethod(currencies, allCurrencyData)
+                    .ifPresent(currencyListBuilder::addMethod);
+
+        }
+
+        TypeSpec currencyListType = currencyListBuilder.build();
+
+        JavaFile javaFile = JavaFile.builder(packageName, currencyListType)
+                .build();
+
+        javaFile.writeTo(pw);
+        pw.close();
+    }
+
+    private Optional<MethodSpec> writeCurrencyMethod(String[] currencies, Map<String, CurrencyInfo> allCurrencyData) {
+
+        if (currencies.length < 1) {
+            return Optional.empty();
+        }
+
+        MethodSpec.Builder loadCurrencyMap = MethodSpec.methodBuilder("loadCurrencyMap");
+        loadCurrencyMap
+                .addModifiers(Modifier.PROTECTED)
+                .returns(ParameterizedTypeName.get(ClassName.get(HashMap.class), TypeName.get(String.class), TypeName.get(CurrencyData.class)))
+                .addStatement("$T<$T,$T> result = super.loadCurrencyMap()", HashMap.class, String.class, CurrencyData.class);
+
+        for (String currencyCode : currencies) {
+            CurrencyInfo currencyInfo = allCurrencyData.get(currencyCode);
+
+            loadCurrencyMap.addComment("$L", currencyInfo.getDisplayName());
+            loadCurrencyMap.addCode(currencyInfo.asMapEntry());
+        }
+        loadCurrencyMap.addStatement("return result");
+        return Optional.of(loadCurrencyMap.build());
+    }
+
+    private Optional<MethodSpec> writeNamesMethod(String[] currencies, Map<String, CurrencyInfo> allCurrencyData) {
+
+        List<CodeBlock> statements = new ArrayList<>();
+
+        for (String currencyCode : currencies) {
+            CurrencyInfo currencyInfo = allCurrencyData.get(currencyCode);
+            String displayName = currencyInfo.getDisplayName();
+            if (displayName != null && !currencyCode.equals(displayName)) {
+                statements.add(CodeBlock.builder()
+                        .addStatement("result.put($S,$S)", currencyCode, displayName).build());
+            }
+        }
+
+        if (!statements.isEmpty()) {
+            MethodSpec.Builder loadNamesMethod = MethodSpec.methodBuilder("loadNamesMap")
+                    .addAnnotation(Override.class)
+                    .addModifiers(Modifier.PROTECTED)
+                    .returns(ParameterizedTypeName.get(ClassName.get(HashMap.class), TypeName.get(String.class), TypeName.get(String.class)))
+                    .addStatement("$T<$T,$T> result = super.loadNamesMap()", HashMap.class, String.class, String.class);
```

Comment:
can also parameterize HashMap on StringString, and just do a single $T, see ParameterizedTypeName.get(...)

Reply:
ok :+1: